prog/postgres/postgres_resource_nexus.rb

```diff
@@ -189,7 +189,20 @@ def before_run
   label def wait
     reap(fallthrough: true)
 
-    if postgres_resource.needs_convergence? && strand.children_dataset.where(prog: "Postgres::ConvergePostgresResource").empty?
+    if postgres_resource.needs_upgrade? && strand.children_dataset.where(prog: "Postgres::UpgradePostgresResource").empty?
+      candidate_server_id = Prog::Postgres::PostgresServerNexus.assemble(
+        resource_id: postgres_resource.id,
+        timeline_id: postgres_resource.timeline.id,
+        timeline_access: "fetch"
+      ).id
+      new_timeline_id = Prog::Postgres::PostgresTimelineNexus.assemble(
+        location_id: postgres_resource.location_id
+      ).id
+
+      bud Prog::Postgres::UpgradePostgresResource, {"candidate_server_id" => candidate_server_id, "new_timeline_id" => new_timeline_id}, :start
```

Comment:
Is there a cleaner way to bud a Prog from here with these values passed in? I thought of .assemble + updating children but that does not seem like a good pattern.